### PR TITLE
Documentation for explicit extrapolation methods

### PR DESCRIPTION
**Pull request by AlthausKonstantin**
Added documentation for ExtrapolationMidpointDeuflhard and ExtrapolationMidpointHairerWanner (cf. https://github.com/JuliaDiffEq/OrdinaryDiffEq.jl/pull/714)

### CODE REVIEW

**Comment by ChrisRackauckas** [comment]
This is a bit wordy and convoluted. Most of these details aren't needed for the docs and should be in the PR instead.

**Comment by ChrisRackauckas** [comment]
I simplified it a bit and put a commit to OrdinaryDiffEq to unify the arg naming.

**Comment by ChrisRackauckas** [comment]
The non-stiff ODE benchmarks from https://github.com/JuliaDiffEq/DiffEqBenchmarks.jl/pull/32 are updated. You might want to make a PR to that branch and see how the extrapolation methods fare on some of those benchmarks.

**Comment by AlthausKonstantin** [comment]
All right, didn't know that I would have gotten away with such brevity. Thanks for going over it!
I will have a look at that PR.
